spec/requests/saved_pages_request_spec.rb

```diff
@@ -0,0 +1,45 @@
+RSpec.describe "SavedPages", type: :request do
+  context "when receiving an unauthenticated request" do
+    it "returns unauthorised for GET /api/saved_pages" do
+      get saved_pages_path
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+  end
+
+  context "when receiving an authenticated request" do
+    before do
+      stub_oidc_discovery
+
+      # rubocop:disable RSpec/AnyInstance
+      allow_any_instance_of(OidcClient).to receive(:tokens!).and_return({ access_token: "access-token", refresh_token: "refresh-token" })
+      # rubocop:enable RSpec/AnyInstance
+
+      FactoryBot.create(:oidc_user)
+    end
+
+    let(:headers) { { "Content-Type" => "application/json", "GOVUK-Account-Session" => session_identifier } }
+    let(:session_identifier) { placeholder_govuk_account_session }
+    let(:user) { OidcUser.find_by(sub: "user-id") }
```

Comment:
Not sure I get the value of creating the user in the before block and then having a let that finds it by `sub`. 🤔
Can't we remove `FactoryBot.create(:oidc_user)` and `let(:user) { OidcUser.find_by(sub: "user-id") }`, and just have `let(:user) { FactoryBot.create(:oidc_user) }`?

Reply:
done!